crates/markdown_preview/src/markdown_parser.rs

```diff
@@ -944,24 +953,111 @@ impl<'a> MarkdownParser<'a> {
         source_range: Range<usize>,
         node: &Rc<markup5ever_rcdom::Node>,
         paragraph: &mut MarkdownParagraph,
+        highlights: &mut Vec<MarkdownHighlight>,
     ) {
+        fn add_highlight_range(
+            text: &String,
+            start: usize,
+            highlights: Vec<MarkdownHighlight>,
+        ) -> Vec<(Range<usize>, MarkdownHighlight)> {
+            highlights
+                .into_iter()
+                .map(|style| (start..text.len(), style))
+                .collect()
+        }
+
         match &node.data {
             markup5ever_rcdom::NodeData::Text { contents } => {
-                paragraph.push(MarkdownParagraphChunk::Text(ParsedMarkdownText {
-                    source_range,
-                    regions: Vec::default(),
-                    region_ranges: Vec::default(),
-                    highlights: Vec::default(),
-                    contents: contents.borrow().to_string().into(),
-                }));
+                let contents = contents.borrow().to_string();
+
+                // append the text to the last chunk, so we can have a hacky version
+                // of inline text with highlighting
+                if let Some(text) = paragraph.iter_mut().last().and_then(|p| match p {
+                    MarkdownParagraphChunk::Text(text) => Some(text),
+                    _ => None,
+                }) {
+                    let mut new_text = text.contents.to_string();
+                    new_text.push_str(&contents);
+                    let highlights = add_highlight_range(
+                        &new_text,
+                        text.contents.len(),
+                        std::mem::take(highlights),
+                    );
+
+                    text.contents = SharedString::from(new_text);
+                    text.highlights.extend(highlights);
+                } else {
+                    paragraph.push(MarkdownParagraphChunk::Text(ParsedMarkdownText {
+                        source_range,
+                        highlights: add_highlight_range(&contents, 0, std::mem::take(highlights)),
+                        regions: Vec::default(),
+                        contents: contents.into(),
+                        region_ranges: Vec::default(),
+                    }));
+                }
             }
             markup5ever_rcdom::NodeData::Element { name, attrs, .. } => {
                 if local_name!("img") == name.local {
                     if let Some(image) = self.extract_image(source_range, attrs) {
                         paragraph.push(MarkdownParagraphChunk::Image(image));
                     }
+                } else if local_name!("b") == name.local || local_name!("strong") == name.local {
+                    highlights.push(MarkdownHighlight::Style(MarkdownHighlightStyle {
+                        italic: false,
```

Comment:
Might be nice to use `..Default::default()` here and only explicitly list the fields that are changed. Would make this test more readable